dev/Prototypes/ItemsViewPrototype/Control/DragDrop/DragDropBehavior.cs

```diff
@@ -0,0 +1,123 @@
+﻿using System.Windows.Input;
+using Windows.ApplicationModel.DataTransfer;
+using Windows.UI.Xaml;
+
+namespace DEPControlsTestApp.ItemsViewPrototype
+{
+    enum DragAndDropOperation
+    {
+        None,
+        Copy,
+        Move,
+    }
+
+    static class DragAndDropBehavior
+    {
+        public static DragAndDropPayload GetSource(UIElement element) => (DragAndDropPayload)element.GetValue(DragAndDropBehavior.SourceProperty);
+        public static void SetSource(UIElement element, DragAndDropPayload value) => element.SetValue(DragAndDropBehavior.SourceProperty, value);
+
+        public static DragAndDropPayload GetTarget(UIElement element) => (DragAndDropPayload)element.GetValue(DragAndDropBehavior.TargetProperty);
+        public static void SetTarget(UIElement element, DragAndDropPayload value) => element.SetValue(DragAndDropBehavior.TargetProperty, value);
+
+
+        public static ICommand GetDropCommand(UIElement element) => (ICommand)element.GetValue(DragAndDropBehavior.DropCommandProperty);
+        public static void SetDropCommand(UIElement element, ICommand value) => element.SetValue(DragAndDropBehavior.DropCommandProperty, value);
+
+        private static void OnTargetPropertyChanged(DependencyObject obj, DependencyPropertyChangedEventArgs args)
+        {
+            var newValue = (DragAndDropPayload)args.NewValue;
+            var oldValue = (DragAndDropPayload)args.OldValue;
+            var element = (UIElement)obj;
+
+            if (oldValue?.Operation != DragAndDropOperation.None)
+            {
+                element.AllowDrop = false;
+                element.DragOver -= DragAndDropBehavior.OnDragOver;
+                element.Drop -= DragAndDropBehavior.OnDrop;
+            }
+
+            if (newValue?.Operation != DragAndDropOperation.None)
+            {
+                element.AllowDrop = true;
+                element.DragOver += DragAndDropBehavior.OnDragOver;
+                element.Drop += DragAndDropBehavior.OnDrop;
+            }
+        }
+
+        private static void OnDragOver(object sender, DragEventArgs e)
+        {
+            var payload = DragAndDropBehavior.GetTarget((UIElement)sender);
+            var operation = payload.Operation.ToDataPackageOperation();
+            var data = payload.Data;
+            e.AcceptedOperation = operation;
+        }
+
+        private static void OnDrop(object sender, DragEventArgs e)
+        {
+            var element = (UIElement)sender;
+            var payload = DragAndDropBehavior.GetTarget(element);
+            var command = DragAndDropBehavior.GetDropCommand((UIElement)sender);
+            command?.Execute(payload.Data);
```

Comment:
Why are we never populating e.Data (the DataPackage field), and then passing 'e' to the command. That would allow the command to have access to the args, and know whether we were dropping onto a move target, or a copy target, and then do the right thing based on that context.

Reply:
I guess I was thinking we could use DataPackage.SetData to set the payload.Data on the e.Data field, and then GetDataAsync off of the DataPackageView, but perhaps that would not work.

Reply:
I messed around, and tried out passing payload, rather than payload.Data, so I would have access to the Target operation type. That seems like a viable option.